src/mainwindow.cpp

```diff
@@ -2189,6 +2189,12 @@ void MainWindow::minimizeWindow()
 void MainWindow::QuitApplication()
 {
     m_noteEditorLogic->closeEditor();
+#if QT_VERSION >= QT_VERSION_CHECK(6, 0, 0)
+    // Works around an issue where trying to exit Notes using the
+    // system tray menu action while our window is hidden, fails.
+    if (MainWindow::isHidden())
+        MainWindow::show();
```

Comment:
Really interesting that this hack works. It makes it somewhat likely to be a Qt issue, and we should probably report it, but in that case we need to create a minimal example app that reproduces the problem.

Reply:
I'll look into creating a minimal example.